main.cpp

```diff
@@ -92,8 +93,90 @@ good to go!
  */
 
 #include <iostream>
+
+struct FloatType
+{
+    float add(float lhs, float rhs)
+    {
+        return lhs + rhs;
+    }
+    float subtract(float lhs, float rhs)
+    {
+        return lhs - rhs;
+    }
+    float multiply(float lhs, float rhs)
+    {
+        return lhs * rhs;
+    }
+    float divide(float lhs, float rhs)
+    {
+        if(rhs == 0.f)
+        {
+            std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+        }
+        return lhs / rhs;
+    }
+
+};
+
+struct DoubleType
+{
+    double add(double lhs, double rhs)
+    {
+        return lhs + rhs;
+    }
+    double subtract(double lhs, double rhs)
+    {
+        return lhs - rhs;
+    }
+    double multiply(double lhs, double rhs)
+    {
+        return lhs * rhs;
+    }
+    double divide(double lhs, double rhs)
+    {
+        if(rhs == 0.)
+        {
+            std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+        }
+        return lhs / rhs;
+    }
+};
+
+struct IntType
+{
+
+    int add(int lhs, int rhs)
+    {
+        return lhs + rhs;
+    }
+    int subtract(int lhs, int rhs)
+    {
+        return lhs - rhs;
+    }
+    int multiply(int lhs, int rhs)
+    {
+        return lhs * rhs;
+    }
+    int divide(int lhs, int rhs)
+    {
+        if(rhs == 0.)
+        {
+            std::cout << "error, integer division by zero will crash the program!" << std::endl;
+            std::cout << "returning lhs" << std::endl;
+            return lhs;
+        }
+        return lhs / rhs;
+    }
+};
+    
 int main() 
 {
+
+/*    I didn't think you'd want this error but it's here
+    
+    std::cout << "clang++ -std=c++17 -Weverything -Wno-c++98-compat -Wno-missing-prototypes main.cpp && ./a.out" << std::endl;
+*/    
```

Comment:
You can remove this comment.  it doesn't need to exist in the project
- [ ] revise